modules/tensor_mechanics/src/materials/StrainEnergyRateDensity.C

```diff
@@ -0,0 +1,65 @@
+//* This file is part of the MOOSE framework
```

Comment:
So it looks like you elected to add an option to EshelbyTensor for the rate form, but add another separate class for the rate form of the strain energy density. I guess I don't have very strong feelings either, way, but we should talk a little about what motivated you to implement it the way you did.

Reply:
Adding an option to `EshelbyTensor` allows for some computation/variable retrieval code reuse, whereas the strain energy density rate is computed in one line of code, which is new. Also, it may be that the `EshelbyTensor` will only be used within the context of the `DomainIntegralAction` so adding an option to it may not mess things up too much. We could eventually want to use the `StrainEnergyRateDensity` for a different application. This second thought is more speculative though.

Reply:
Makes sense.